File: app/src/main/java/me/mitul/aij/reg/RegisterActivity.java
package me.mitul.aij.reg;

import android.app.AlertDialog;
import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.provider.Settings;
import android.text.TextUtils;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.AutoCompleteTextView;

import androidx.appcompat.app.AppCompatActivity;

import me.mitul.aij.helper.HelperRegistration;
import me.mitul.aij.R;

public class RegisterActivity extends AppCompatActivity {
    private AutoCompleteTextView editTextFullName;
    private AutoCompleteTextView editTextEmail;
    private AutoCompleteTextView editTextMobile;
    private AutoCompleteTextView editTextCity;
    private AutoCompleteTextView editTextPassword;
    private AutoCompleteTextView editTextPasswordRe;
    private View progressView;
    private View regFormView;
    private Animation shake;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        editTextFullName = findViewById(R.id.full_name_reg);
        editTextEmail = findViewById(R.id.email_reg);
        editTextMobile = findViewById(R.id.mobile_reg);
        editTextCity = findViewById(R.id.city_reg);
        editTextPassword = findViewById(R.id.password_reg);
        editTextPasswordRe = findViewById(R.id.password_reg_re);
        progressView = findViewById(R.id.fab_register_go);
        regFormView = findViewById(R.id.reg_form);

        findViewById(R.id.fab_register_go).setOnClickListener(view -> attemptReg());
        findViewById(R.id.fab_skip_go).setOnClickListener(v -> {
            new AlertDialog.Builder(RegisterActivity.this)
                    .setMessage("Registration required!")
                    .setCancelable(false)
                    .setPositiveButton("OK", null)
                    .create().show();
        });
        shake = AnimationUtils.loadAnimation(RegisterActivity.this, R.anim.shake);
    }

    private void attemptReg() {
        boolean isValid = true;
        String strTextFullName = editTextFullName.getText().toString();
        String strTextEmail = editTextEmail.getText().toString();
        String strTextMobile = editTextMobile.getText().toString();
        String strTextCity = editTextCity.getText().toString();
        String strTextPassword = editTextPassword.getText().toString();
        String strTextPasswordRe = editTextPasswordRe.getText().toString();
        String deviceId = Settings.System.getString(RegisterActivity.this.getContentResolver(), Settings.Secure.ANDROID_ID);

        if (TextUtils.isEmpty(strTextFullName)) {
            editTextFullName.startAnimation(shake);
            isValid = false;
        }
        if (TextUtils.isEmpty(strTextEmail) || !strTextEmail.contains("@") || !strTextEmail.contains(".")) {
            editTextEmail.startAnimation(shake);
            isValid = false;
        }
        if (TextUtils.isEmpty(strTextMobile)) {
            editTextMobile.startAnimation(shake);
            isValid = false;
        }
        if (TextUtils.isEmpty(strTextCity)) {
            editTextCity.startAnimation(shake);
            isValid = false;
        }
        if (TextUtils.isEmpty(strTextPassword) ||
                strTextPassword.length() < 6) {
            editTextPassword.startAnimation(shake);
            isValid = false;
        }
        if (TextUtils.isEmpty(strTextPasswordRe) ||
                !strTextPassword.equals(strTextPasswordRe)) {
            editTextPasswordRe.startAnimation(shake);
            isValid = false;
        }
        if (isValid) {
            new UserRegTask().execute(strTextFullName, strTextEmail, strTextMobile, strTextCity, strTextPassword, strTextPasswordRe, deviceId);
        }
    }

    private class UserRegTask extends AsyncTask<String, Void, Boolean> {
        String email;
        String pass;

        @Override
        protected Boolean doInBackground(String... strings) {
            try {
                Thread.sleep(2000);
                return new HelperRegistration(RegisterActivity.this)
                        .attemptReg(email = strings[0], strings[1], strings[2], strings[3], pass = strings[4], strings[5], strings[6]);
            } catch (InterruptedException ignored) {
            }
            return false;
        }

        @Override
        protected void onPreExecute() {
            progressView.startAnimation(AnimationUtils.loadAnimation(getApplicationContext(), R.anim.anim_fab));
        }

        @Override
        protected void onPostExecute(Boolean success) {
            if (success) {
                Intent intent = new Intent(RegisterActivity.this, LoginActivity.class);
                intent.putExtra("IsRegistered", true)
                        .putExtra("UserName", email)
                        .putExtra("Password", pass);
                startActivity(intent);
                finish();
            } else {
                regFormView.startAnimation(shake);
            }
        }
    }
}

File: app/src/main/java/me/mitul/aij/reg/RegisterActivity.kt
package me.mitul.aij.reg

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.text.TextUtils
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.AutoCompleteTextView
import android.widget.LinearLayout
import com.google.android.material.floatingactionbutton.FloatingActionButton
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.coroutineScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import me.mitul.aij.R
import me.mitul.aij.helper.HelperRegistration
import me.mitul.aij.home.HomeScreenActivity
import me.mitul.aij.utils.Consts
import kotlin.coroutines.EmptyCoroutineContext

class RegisterActivity : Activity() {
    private lateinit var tvUsername: AutoCompleteTextView
    private lateinit var tvEmail: AutoCompleteTextView
    private lateinit var tvPhone: AutoCompleteTextView
    private lateinit var tvCity: AutoCompleteTextView
    private lateinit var tvPassword: AutoCompleteTextView
    private lateinit var tvRePassword: AutoCompleteTextView

    private lateinit var fabRegister: FloatingActionButton
    private lateinit var registerForm: LinearLayout

    private lateinit var shake: Animation
    private lateinit var rotate: Animation

    private lateinit var intentLogin: Intent
    private lateinit var dbHelper: HelperRegistration

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        intentLogin = Intent(applicationContext, LoginActivity::class.java)
        dbHelper = HelperRegistration(applicationContext)

        shake = AnimationUtils.loadAnimation(applicationContext, R.anim.shake)
        rotate = AnimationUtils.loadAnimation(applicationContext, R.anim.anim_fab)

        tvUsername = findViewById(R.id.reg_ed_username)
        tvEmail = findViewById(R.id.reg_ed_email)
        tvPhone = findViewById(R.id.reg_ed_phone)
        tvCity = findViewById(R.id.reg_ed_city)
        tvPassword = findViewById(R.id.reg_ed_password)
        tvRePassword = findViewById(R.id.reg_ed_re_password)
        registerForm = findViewById(R.id.registration_form)

        fabRegister = findViewById<FloatingActionButton>(R.id.reg_fab_register).also {
            it.setOnClickListener {
                CoroutineScope(EmptyCoroutineContext).launch { attemptRegister() }
            }
        }
        findViewById<FloatingActionButton>(R.id.reg_fab_login).setOnClickListener {
            startActivity(intentLogin)
            finish()
        }
    }

    private suspend fun attemptRegister() {
        val username = tvUsername.text.toString()
        val email = tvEmail.text.toString()
        val phone = tvPhone.text.toString()
        val city = tvCity.text.toString()
        val password = tvPassword.text.toString()
        val rePassword = tvRePassword.text.toString()

        if (TextUtils.isEmpty(username)) {
            tvUsername.startAnimation(shake)
            return
        }
        if (TextUtils.isEmpty(email) || !email.contains("@") || !email.contains(".")) {
            tvEmail.startAnimation(shake)
            return
        }
        if (TextUtils.isEmpty(phone)) {
            tvPhone.startAnimation(shake)
            return
        }
        if (TextUtils.isEmpty(city)) {
            tvCity.startAnimation(shake)
            return
        }
        if (TextUtils.isEmpty(password) || password.length < 6) {
            tvPassword.startAnimation(shake)
            return
        }
        if (TextUtils.isEmpty(rePassword) || password != rePassword) {
            tvRePassword.startAnimation(shake)
            return
        }

        fabRegister.startAnimation(rotate)

        coroutineScope {
            try {
                delay(timeMillis = 2000L)
                val deviceId =
                    Settings.System.getString(contentResolver, Settings.Secure.ANDROID_ID)
                val status = dbHelper.register(username, email, phone, city, password, deviceId)
                if (status != -1L) {
                    intentLogin.putExtra(Consts.KEY_IS_REGISTERED, true)
                        .putExtra(Consts.KEY_USERNAME, username)
                        .putExtra(Consts.KEY_PASSWORD, password)
                    startActivity(intentLogin)
                    finish()
                } else {
                    registerForm.startAnimation(shake)
                }
            } catch (ignored: InterruptedException) {
            }
        }
    }
}
